addons/website_sale_autocomplete/controllers/main.py

```diff
@@ -0,0 +1,151 @@
+                if field_standard == 'country':
+                    standard_data[field_standard] = request.env['res.country'].search(
+                        [('code', '=', google_field['short_name'].upper())])[0].id
+                elif field_standard == 'state':
+                    state = request.env['res.country.state'].search(
+                        [('code', '=', google_field['short_name'].upper()),
+                         ('country_id.id', '=', standard_data['country'])])
+                    if len(state) == 1:
+                        standard_data[field_standard] = state.id
+                else:
+                    standard_data[field_standard] = google_field['long_name']
+        return standard_data
+
+    def _guess_number_from_input(self, source_input, standard_address):
+        """
+        Google might not send the house number in case the address
+        does not exist in their database.
+        We try to guess the number from the user's input to avoid losing the info.
+        """
+        # Remove other parts from address to make better guesses
+        guess = source_input.replace(standard_address.get('zip', ''), '') \
+            .replace(standard_address.get('street', ''), '') \
+            .replace(standard_address.get('city', ''), '')
+        guess = guess.split(',')[0].strip()
+        return guess
+
+    def _perform_place_search(self, partial_address, session_uuid=None, language_code=None, country_code=None):
+        params = {
+            'key': self._get_key(),
+            'fields': 'formatted_address,name',
+            'inputtype': 'textquery',
+            'types': 'address',
+            'input': partial_address
+        }
+        if country_code:
+            params['components'] = f'country:{country_code}'
+        if language_code:
+            params['language'] = language_code
+        if session_uuid:
+            params['sessiontoken'] = session_uuid
+
+        try:
+            results = requests.get(f'{GOOGLE_PLACES_ENDPOINT}/autocomplete/json', params=params, timeout=TIMEOUT).json()
+        except TimeoutError:
+            return []
+        results = results['predictions']
+
+        # Convert google specific format to standard format.
+        return [{
+            'formatted_address': result['description'],
+            'google_place_id': result['place_id'],
+        } for result in results]
+
+    def _perform_complete_place_search(self, address, google_place_id=None, language_code=None, session_uuid=None):
+        params = {
+            'key': self._get_key(),
+            'place_id': google_place_id,
+            'fields': 'address_component,adr_address'
+        }
+
+        if language_code:
+            params['language'] = language_code
+        if session_uuid:
+            params['sessiontoken'] = session_uuid
+
+        try:
+            results = requests.get(f'{GOOGLE_PLACES_ENDPOINT}/details/json', params=params, timeout=TIMEOUT).json()
```

Comment:
@odony 
And here is the second requests call. Mostly the same as the first one so I won't repeat the same comment